CallAutomation_OutboundCalling/src/app.ts

```diff
@@ -0,0 +1,195 @@
+import { config } from 'dotenv';
+import fs from "fs";
+import express, { Application } from 'express';
+import { PhoneNumberIdentifier } from "@azure/communication-common";
+import { CallAutomationClient, CallConnection, CallMediaRecognizeDtmfOptions, CallLocator, StartRecordingOptions, FileSource, CallInvite } from "@azure/communication-call-automation";
+import path from 'path';
+
+config();
+
+const PORT = process.env.PORT;
+const app: Application = express();
+app.use(express.static('webpage'));
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+
+let callConnectionId: string;
+let recordingId: string;
+let callConnection: CallConnection;
+let serverCallId: string;
+let callee: PhoneNumberIdentifier;
+let acsClient: CallAutomationClient;
+
+async function createAcsClient() {
+	const connectionString = process.env.CONNECTION_STRING || "";
+	acsClient = new CallAutomationClient(connectionString);
+	console.log("Initialized ACS Client.");
+}
+
+async function createOutboundCall() {
+	const callInvite: CallInvite = {
+		targetParticipant: callee,
+		sourceCallIdNumber: {
+			phoneNumber: process.env.ACS_RESOURCE_PHONE_NUMBER || "",
+		},
+	};
+
+	console.log("Placing outbound call...");
+	acsClient.createCall(callInvite, process.env.CALLBACK_URI + "ongoingcall");
+}
+
+async function startRecording() {
+	try {
+		const callLocator: CallLocator = {
+			id: serverCallId,
+			kind: "serverCallLocator",
+		};
+
+		const recordingOptions: StartRecordingOptions = {
+			callLocator: callLocator,
+		};
+
+		const response = await acsClient.getCallRecording().start(recordingOptions);
+
+		recordingId = response.recordingId;
+	} catch (error) {
+		console.error("Error starting recording:", error);
+		throw error;
+	}
+}
+
+async function playAudio(prompt: string) {
+	try {
+		const audioPrompt: FileSource[] = [{
+			url: process.env.MEDIA_CALLBACK_URI + prompt,
+			kind: "fileSource",
+		}];
+
+		await callConnection.getCallMedia().playToAll(audioPrompt);
+	} catch (error) {
+		console.error('An error occurred while playing audio prompt:', error);
+	}
+}
+
+async function startToneRecognition() {
+	try {
+		const audioPrompt: FileSource = {
+			url: process.env.MEDIA_CALLBACK_URI + "MainMenu.wav",
+			kind: "fileSource",
+		};
+
+		const recognizeOptions: CallMediaRecognizeDtmfOptions = {
+			playPrompt: audioPrompt,
+			kind: "callMediaRecognizeDtmfOptions",
+		};
+
+		await callConnection.getCallMedia().startRecognizing(callee, 1, recognizeOptions);
+	} catch (error) {
+		console.error("Error starting tone recognition:", error);
+		throw error;
+	}
+}
+
+async function hangUpCall() {
+	await acsClient.getCallRecording().stop(recordingId);
+	callConnection.hangUp(true);
+}
+
+// POST endpoint to handle incoming call events
+app.post("/incomingcall", async (req: any, res: any) => {
```

Comment:
This is not needed for outbound calling scenario.

Reply:
dont they still need to set up the callback uri to receive events from ACS?